Publicar introducao-jupyter-notebooks

DanielAlvesLABDH opened the pull request:
Publicar a tradução da lição "Introduction to Jupyter Notebooks"

### Checklist

- [x] Assign yourself in the "Assignees" menu
- [x] Add the appropriate "Label"
- [ ] If this PR closes an Issue, add the phrase `Closes #ISSUENUMBER` to your summary above
- [ ] Ensure the status checks pass: if you have difficulty fixing build errors, please contact our Publishing Assistant @anisa-hawes 
- [x] Check the Netlify Preview: navigate to netlify/ph-preview/deploy-preview and click 'details' (at right)
- [x] Assign at least one individual or team to "Reviewers"

DanielAlvesLABDH committed:
Publicar introducao-jupyter-notebooks

Publicar a tradução da lição "Introduction to Jupyter Notebooks"

DanielAlvesLABDH committed:
Add files via upload

Adicionar imagens da lição

DanielAlvesLABDH committed:
Update introducao-jupyter-notebooks.md

Corrigir cabeçalho

anisa-hawes committed:
Update introducao-jupyter-notebooks.md

Add DOI

ericbrasiln commented:
Hi, @DanielAlvesLABDH 

I will finish the review by tomorrow.

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
```diff
@@ -0,0 +1,427 @@
+
+## Navegando na interface do Jupyter Notebook
+
+A interface do gerenciador de ficheiro do Jupyter Notebook é a principal maneira de abrir um ficheiro Jupyter Notebook (.ipynb). Se você tentar abrir um em um editor de texto simples, o notebook será exibido como um ficheiro JSON, não com blocos interativos de código. Para visualizar um notebook através da interface Jupyter, você tem que abrir o Jupyter Notebook primeiro (que será exibido em uma  ...
```
Corrigir "Se você tentar abrir um em um editor de texto simples" para "Se você tentar abrir em um editor de texto simples"

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
```diff
@@ -0,0 +1,427 @@
+Para evitar desordenar esta pasta, você pode fazer uma nova pasta dentro deste diretório para seus notebooks. Você pode fazer isso na sua interface usual de gerenciamento de ficheiro(s)(Finder no Mac, ou File Explorer no Windows), ou dentro do próprio Jupyter Notebook, já que o Jupyter Notebook, assim como o Google Drive, fornece uma interface de gerenciamento de ficheiro(s) dentro de um navegado ...
+
+## Upload dos dados do exemplo
+O ficheiro CSV de exemplo para esta lição é um extrato de metadados de fan fiction de Harry Potter coletados do site de fanfic italiano https://efpfanfic.net, depois limpos usando uma combinação de [expressões regulares](/en/lessons/understanding-regular-expressions) e [OpenRefine](/pt/licoes/limpar-dados-openrefine). O CSV tem três colunas: a classificação da histó_amarelo_ (amarelo), _laranja_  ...
```
Corrigir:
"O CSV tem três colunas: a classificação da histó_amarelo_ (amarelo), _laranja_ (laranja) e _veermelho_ (vermelho)."

Por:
" a classificação da história (similar a uma classificação de filme), a data que foi originalmente publicada, e a data mais recente de atualização. As opções de classificação são _verde_ (verde), _giallo_ (amarelo), _arancione_ (laranja), e _rosso_ (vermelho)."

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+
+Voltando ao nosso exemplo, em seguida adicione uma nova célula de código e cole o seguinte código (certifique-se de que incluiu os espaçamentos):
+  
+```
````
Nesse bloco de código é preciso corrigir a indetação e adicionar um ")" ao final da 4ª linha. Deve ficar assim:

```
    with  open('ph-jupyter-notebook-example.csv') as f:
         csv_reader = csv.reader(f, delimiter=',')
         for row in csv_reader:
              datetime.datetime.strptime(row[1], '%d/%m/%Y').strftime('%A')
              print(row)

```

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
```diff
@@ -0,0 +1,427 @@
+
+As duas abordagens abrirão uma nova janela ou guia no seu navegador padrão com a interface Jupyter Notebook. O Jupyter Notebook é baseado no navegador: você só interage com ele através do seu navegador, mesmo quando o Jupyter Notebook está sendo executado no seu próprio computador.
+
+Se você estiver usando notebooks que importam pacotes Python que têm dependências de versões específicas de outros pacotes, você deve configurar um ambiente para usar com esses notebooks, para que você não lide com conflitos de versão (por exemplo, se um notebook requer a versão 1.0 de um pacote, e outro requer a versão 2.0). [A documentação do Anaconda Navegador para Gerenciar Ambientes](https:/ ...
```
Esse parágrafo tem um destaque na versão versão original. Colocá-lo entre:
```
<div class="alert alert-warning">

</div>
```

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+ 
+Execute as duas células de código no notebook, começando pelo topo.
+  
+```
````
remover

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+Execute as duas células de código no notebook, começando pelo topo.
+  
+```
+O Jupyter notebook funciona melhor se você executar as células sequencialmente. Às vezes, você pode obter erros ou saídas incorretas se executar as células fora de ordem ou tentar editar e executar iterativamente diferentes partes do notebook. Se você fez muitas alterações e executou blocos de código de forma não linear e descobrir que você está recebendo uma saída estranha, você pode redefinir o ...
````
Esse parágrafo tem um destaque na versão versão original. Colocá-lo entre:
```
<div class="alert alert-warning">

</div>
```

ericbrasiln replied:
E como está dentro de uma `<div>`, a ênfase itálico deve usar o padrão `<i>texto</i>`

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+```
+O Jupyter notebook funciona melhor se você executar as células sequencialmente. Às vezes, você pode obter erros ou saídas incorretas se executar as células fora de ordem ou tentar editar e executar iterativamente diferentes partes do notebook. Se você fez muitas alterações e executou blocos de código de forma não linear e descobrir que você está recebendo uma saída estranha, você pode redefinir o ...
+
+```
````
remover

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+
+```
+# Fanfic date conversion
+Converting published & updated dates for Italian fanfic into days of the week.
````
Traduzir para o português

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+Abaixo dele, adicione uma nova célula de código com o seguinte código (prestando atenção ao espaçamento, de modo que o código seja indentado assim como você vê abaixo):
+ 
+ 
+```
````
Corrigir a indetação desse bloco de código para:

```
with open('ph-jupyter-notebook-example.csv') as f:
  csv_reader = csv.reader(f, delimiter=',')
  for row in csv_reader:
        parseddate = dateutil.parser.parse(row[1])
        print(parseddate)

 ```

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
````diff
@@ -0,0 +1,427 @@
+
+Se você não quiser resolver isso por conta própria, você pode copiar e colar esse código em uma nova célula de código ou substituir a célula de código atual:
+
+```
````
Corrigir indetação do bloco para:

```
#identifies the source file to open, calls it f
with open('ph-jupyter-notebook-example.csv') as f:
  #creates an output file (referred to as "out" in the notebook) for you to write to
 with open('ph-jupyter-notebook-example-dayofweek.csv', 'w') as out:
    #defines "csv_reader" as running the function csv.reader on the file
    csv_reader = csv.reader(f, delimiter=',')
    #defines "csv_writer" as running the functin csv.writer to "out" (the output file)
    csv_writer = csv.writer(out)
    #for each row that's being read by csv_reader...
    for row in csv_reader:
        #defines "csv_reader" as running the function csv.reader on the file
        csv_reader = csv.reader(f, delimiter=',')
        #for each row that's being read by csv_reader...
        for row in csv_reader:
            #creates a list called "values" with the contents of the row
            values = list(row)
            #defines "rating" as the first thing in the list
            #counting in Python starts with 0, not 1
            rating = values[0]
            #defines "parseddatepub" as the second thing (1, because we start with 0) in the list,
            #converted into a standard date format using dateutil.parser
            #and when those dates are parsed, the parser should know
            #that the first value in the sequence is the day
            parseddatepub = dateutil.parser.parse(values[1], dayfirst=True)
            #same as above for the updated date, the third thing (2) in the list
            parseddateupdate = dateutil.parser.parse(values[2], dayfirst=True)
            #defines "dayofweekpub" as parseddatepub (defined above), converted to the day of week
            #%A is what you use to change it to the day of the week
            #You can see othe formats here: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior
            dayofweekpub = datetime.date.strftime(parseddatepub, '%A')
            #same thing for update date
            dayofweekupdate = datetime.date.strftime(parseddateupdate, '%A')
            #creates a list of the rating and the newly formatted dates
            updatedvalues = [rating, dayofweekpub, dayofweekupdate]
            #writes all the values under this code cell
            csv_writer.writerow(updatedvalues)
            print(updatedvalues)
```

ericbrasiln replied:
Os comentários do código também não devem ser traduzidos?

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
```diff
@@ -0,0 +1,427 @@
+
+Como outra abordagem, você também pode usar Jupyter notebooks para escrever código na medida em que o desenvolve. Em tal oficina, os alunos podem começar com um caderno em branco, e escrever o código junto com você. As células ajudam a segmentar o código como você o escreve, em vez de usar um editor de texto ou IDE (Ambiente de Desenvolvimento Integrado) que não quebra o código de forma tão clara ...
+
+Você pode usar Jupyter notebooks para tarefas em sala de aula dando instruções em Markdown e fazendo com que os alunos escrevam código em uma cela em branco com base nas instruções. Dessa forma, você pode criar uma tarefa de programação interativa que ensina aos alunos não apenas a sintaxe e o vocabulário de uma linguagem de programação, mas também pode explicar as melhores práticas de programaçã ...
```
Corrigir 'cela' para 'célula'

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
```diff
@@ -0,0 +1,427 @@
+
+Se você já está usando Jupyter notebooks para documentar o fluxo de trabalho do seu projeto, você pode ser capaz de reformular esses cadernos de pesquisa para uso em sala de aula, como uma maneira de trazer sua pesquisa para a sala de aula. Este [exemplo de caderno pedagógico](/assets/jupyter-notebooks/ph-jupyter-notebook-example.ipynb) é um híbrido de algumas das abordagens pedagógicas descritas ...
+
+Existem muitos cursos e workshops de 'Introdução ao Python' nas Humanidades muitos Digitais que utilizam Jupyter Notebook (incluindo [Introdução ao Python e Desenvolvimento web com Python para as Humanidades](https://perma.cc/ANL2-K7SM) by Thibault Clérice, traduzido do material por Matthew Munson). O Jupyter Notebook também é comumente usado em oficinas de análise de texto, como a [oficina de ve ...
```
remover 'muitos' entre humanidades digitais.

ericbrasiln commented on pt/licoes/introducao-jupyter-notebooks.md:
```diff
@@ -0,0 +1,427 @@
+
+## Dimensionando a computação com Jupyter Notebooks
+
+Especialmente se você é novo em programar em Python, apenas conseguir qualquer coisa para trabalhar pode parecer uma vitória. No entanto, se você começar a trabalhar com conjuntos de dados maiores, poderá descobrir que algumas das “soluções” iniciais encontradas (como usar `readlines()` para ler um ficheiro de texto linha por linha) se tornam computacionalmente ineficientes, a ponto de causar pro ...
```
corrigir: 'aordenação' para 'a ordenação'

ericbrasiln reviewed (changes requested):
@DanielAlvesLABDH 
Finalizei a revisão. Fiz os comentários diretamente nas linhas.

DanielAlvesLABDH committed:
Update introducao-jupyter-notebooks.md

@ericbrasiln, muito obrigado pela revisão super atenta. E desculpe a demora em responder. Julgo que implementei todas as revisões. Abraço

DanielAlvesLABDH commented:
Dear @anisa-hawes, the PR is failing because of two links in /en/lessons/introduction-to-populating-a-website-with-api-data.html and /en/supporters.html. Can you help, please?

Commit:
Merge branch 'gh-pages' into Publicar_introducao-jupyter-notebooks

anisa-hawes commented:
Hello @DanielAlvesLABDH. This is ready. I've fixed the links in #2979. In the end, only one of the two link failures identified by the build check needed updating (the other was a temporary failure). 

If all of Eric's change requests are complete, then you can click to Resolve Conversation. Is there anything else I can help you with here?

DanielAlvesLABDH commented:
Thanks a lot @anisa-hawes! Let's wait for @ericbrasiln to check if my changes were good and then we publish.

ericbrasiln reviewed (approved):
Aprovado!
